### PR TITLE
feat: add passenger and rain sounds

### LINKED ISSUE #6055: Rain is not audible in cockpit
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-10-21T07:55:04+00:00",
  "ref": "master",
  "sha": "d3754f4b5ee5f6b6f44eaa787e0ec4c27f105748",
  "actor": "davidwalschots",
  "event_name": "manual"
}
```

### Describe the bug

When it's raining, you don't hear it in the cockpit

### Expected behavior

I think it should be audible in the cockpit but I'm not a RL 320neo pilot

### Steps to reproduce

1. Spawn in a place where it's raining
2. with everything off, I think you should be able to hear the rain 

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

_No response_

### PR DESCRIPTION
**Pull request by hotshotp**
Fixes #6055 and #6492

## Summary of Changes

This PR adds a bunch of long awaited features!

- Adds rain sounds that are dynamic   (@mico975)
- Improves thrust reverse sounds (interior @mico975)
- Improve radio altimeter call out sound reverb (@mico975)
- Reworked exterior sound attenuation (@mico975)
- Added passenger ambience and announcement sounds (@hotshotp)
- Added EFB option for passenger ambience and announcement sounds (@ErickSharp)

## Additional context

Discord username (if different from GitHub):

Ping me on discord boris#8008 or Mico#3145 for any questions or concerns, or if unsure about something about the PR in general.

## Testing instructions

Load into aircraft, and check that the efb options outlined in the description are there. Enable rainy weather and observe the sound. Turn the aircraft on and board some passengers from the W&B section of the ATSU and listen for the sound of passengers and boarding music in the cabin. Close and open cockpit door and hear the change. Check and uncheck the EFB option to make sure the sounds stop. When boarding finishes, listen for the "boarding completed" announcement. 30 seconds later, the captain will speak and welcome the passengers. While the announcement happens, enable and disable the EFB announcements option and check that it stops it. Turn the apu and the apu bleed on and listen outside for the packs and the general ambience, and the new attenuation curves. When the captain is done speaking, turn the beacon light on which should prompt the "flight attendants, arm doors and xcheck" announcement, and 30 seconds later the safety briefing. Start as normal. Take off and listen to the rain, it should sound progressively more loud and should change according to wind gusts and rain changes. After taking off, try to sneak in some flybys with drone camera to check for new attenuation curves. Upon reaching cruise, after about 30 seconds the captain will make an announcement. Upon descent, about 30 seconds later the captain will make another announcement. On approach try to do a few flybys to listen to new exterior attenuation curves. Also on approach listen to new radio altimeter callouts and the new reverb. Upon landing use full reverse to listen to the new sample. When shutting down engines, listen in cabin for seatbelts unclicking and when turning off beacon light, doors disarm announcement after a few seconds.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/instruments/src/EFB/Settings/Settings.tsx [hunk 1]
```diff
@@ -813,6 +815,18 @@ const AudioPage = () => {
                     <Slider className="w-60" value={windVolume + 50} onInput={(value) => setWindVolume(value - 50)} />
                 </div>
             </div>
+            <div className="py-4 flex flex-row justify-between items-center">
+                <span className="text-lg text-gray-300">Passenger Ambience</span>
+                <div className="flex flex-row items-center py-1.5">
+                    <Toggle value={!!passengerAmbienceEnabled} onToggle={(value) => setPassengerAmbienceEnabled(value ? 1 : 0)} />
```
Comment (tracernz) [inline src/instruments/src/EFB/Settings/Settings.tsx:821]: Shouldn't that be `value ? 0 : 1`?

Comment (ErickSharp) [reply src/instruments/src/EFB/Settings/Settings.tsx:821]: No

Comment (ErickSharp) [reply src/instruments/src/EFB/Settings/Settings.tsx:821]: The Toggle component provides you with the new value.

Comment (tracernz) [reply src/instruments/src/EFB/Settings/Settings.tsx:821]: Ok

**Review by tracernz — APPROVED** [review]

**Comment by Samueleonard** [comment]
Quality Assurance Tester Report

Discord: 1samueleonard1
Object of testing: (#6732)
Tier of Testing: (1)
Date: (22/02/2022)

Testing Process: 
. loaded in c+d

ANNOUNCEMENT TESTING
. loaded efb + checked options - all there and working
. setup plane as normal - did not start boarding
. started boarding, checked with cockpit door etc, sounds played
. turned on beacon - arm doors announcement plays
. delay after arming doors - safety announcement plays
. climbed to cruise, decended as normal etc - all announcements seemed to play

OTHER TESTING
. set in rainy conditions - listen to new rain effect 😍 
. new GPWS sounds are much better - i asked about this a few times so this was big for me

Negatives: n/a
Testing Results: (Passed)

Conclusions: really nice PR, adds some very cool QOL features to the airplane and with some further enhancement in future PRs, such as additional airlines and further logic for emergencies etc, it could be even  more powerful
Media: n/a

**Comment by St54Kevin** [comment]
**Quality Assurance Tester**

Object of testing: #6732 
Tier of Testing: 2
Date: 22/02/2022

**Testing Process:**
1. Loaded and powered up, checked EFB settings
2. Turned on Rain, see negatives
3. Boarded PAX, see negatives
4. Tested door, see negatives 
5. Option turns it off
6. Packs sound good
7. Turned on beacon
8. taxi and takeoff in rain, inflight rain sounds great I like it
9. Landed and payed attention to RA callouts and reverse

**Negatives:** 
- Rain could be a little louder at the gate, With "Rain" preset there was NO sound. Just with "Storm" preset. Also while standing it still sounds alot like an outside recording. We are still in a metal tube :)
- Boarding sound sounds more like you are sitting in the middle of the cabin rather than in the cockpit. I find it a little immersion braking :( Its too detailed and clear, remember there is still a wall between you and the cabin. It is also not directional. It should come from the cockpit door. Turning the head doesnt change direction its kinda centered. You can hear its a recording from WITHIN the cabin, thus it should come from behind. If you make it come from the "headset" it should more sound like its not from the cabin, more from the "system". Currently the "speaker sound" is very apparent.
- I think the ambience sound needs to get quieter, I may have mistaken the boarding sound for this one.
- That sneezing is REALLY annoying :D 
- Closing the door completely shuts down the boarding sound. IRL you can still hear footspteps and all those low freq sounds quite well. It should be heavily damped rather than shut down
- When cockpit door is closed there is no announcement sound. IRL you can sill hear it a LITTLE, why not binding this to the PA reception knob of the ACP?
- The safety demo announcement is extremely short, where is that taken from? :D

**Testing Results:** Just Passed, I think it needs alot of improvements, which might also come in a later iteration, I personally would not merge it until we have audio silders for the sounds

**Comment by daanvb** [comment]
I’m not a tester, just reading the comments but please make that sneezing/coughing/random bodily ablutions optional. I disabled these in XPRealistic as they are very off putting! Fine as a bit of a gimmick but not for any long term use!

**Comment by hotshotp** [comment]
Thanks for the feedback! I'll be sure to check everything that was said and make improvements.

**Comment by hotshotp** [comment]
> I’m not a tester, just reading the comments but please make that sneezing/coughing/random bodily ablutions optional. I disabled these in XPRealistic as they are very off putting! Fine as a bit of a gimmick but not for any long term use!

no worries, i just had a sneeze in the loop which i easily can fix.

**Comment by davidnorfolk96** [comment]
Quality Assurance Report

Discord: !davidn123#0001
Object of testing: 6732
Tier of Testing: 1
Date: 22/0/22

Testing Process: Loaded in C&D and tested the instructions given by Boris.  Loved the new rain sounds too.  
Negatives: N/A
Testing Results: Passed

Conclusions: N/A, Nice job on the new sounds for the cabin, Feels a little more immersive. I would have to agree with having some kind of audio sliders. Everyone has their own opinion on how they should sound. I think having different audio settings will cater to everyone.

**Comment by hotshotp** [comment]
I have just fixed all of the issues listed by @St54Kevin - thanks for the complex feedback! All is done and ready to test and subsequently merge.

**Comment by mico975** [comment]
Quick check of latest passengers related changes... LGTM

**Pull request merged**